pkg/accountmanager/client.go

```diff
@@ -11,48 +11,43 @@ import (
 	"github.com/aws/aws-sdk-go/aws/session"
 )
 
+//go:generate mockery -name clienter
 type clienter interface {
-	Config(roleArn *arn.ARN) *aws.Config
+	Config(roleArn *arn.ARN, roleSessionName string) *aws.Config
 	IAM(roleArn *arn.ARN) iamiface.IAMAPI
 }
 
+// Default IAM session name for operations performed by the DCE system
+const systemSessionName = "DCESystem"
+
 // Client helps with client management testing and abstraction
 type client struct {
 	session *session.Session
 	sts     stsiface.STSAPI
-	configs map[string]*aws.Config
 }
 
 // Config configures caching of credentials
-func (c *client) Config(roleArn *arn.ARN) *aws.Config {
+func (c *client) Config(roleArn *arn.ARN, roleSessionName string) *aws.Config {
 
 	// return no config for nil inputs
 	if roleArn == nil {
 		return nil
 	}
 
-	key := roleArn.String()
-
-	// check for cached config
-	if c.configs != nil && c.configs[key] != nil {
```

Comment:
Not sure what's going on here, but this caching behavior seems to be causing the functional tests to fail, after their third run. I did try adding the new roleSessionName parameter to the cache key, but it didn't help.

I would run functional tests three times. In the third run (and all subsequent runs) the `POST /leases/auth` endpoint would return what _looked_ like valid credentials, but when you went to use them you'd get an _Invalid Credentials_ error from AWS.
I've been running functional tests over and over locally, and removing this caching logic seems to give me consistent success.